tests/test_sync.py

```diff
@@ -218,18 +226,9 @@ def test_diff_with_editable(fake_dist, from_editable):
         ['django==1.8', 'click==4.0'],
     ]
 )
-def test_sync_install(from_line, lines):
+def test_sync_install(from_line, lines, mocked_tmp_req):
     with mock.patch('piptools.sync.check_call') as check_call:
         to_install = {from_line(line) for line in lines}
 
         sync(to_install, set())
-        check_call.assert_called_once_with(['pip', 'install', '-q'] + sorted(lines))
-
-
-def test_sync_with_editable(from_editable):
```

Comment:
I'm not sure I understand why this test was removed. Can you explain?

Reply:
Because we don't install packages by `pip install -e git+https://...` anymore.

Reply:
Is that a feature that is safe to remove? What happens to users using that feature? For example, the users from #493.

There are a number of areas that still reference editable requirements. If we do remove this, does that require cleanup too? Here is just one example:

https://github.com/jazzband/pip-tools/blob/b558fc86fa128c1a19fc4efee37df4c5031eced8/piptools/sync.py#L74-L75

Reply:
Oh! In the temporary `requirements.txt`, the requirement will include a `-e`, is that right? If so, got it. Maybe an explicit test for this case could be included alongside `test_format_requirement_ireq_with_hashes`.

Reply:
> Oh! In the temporary `requirements.txt`, the requirement will include a `-e`, is that right?

That's right! 

> If so, got it. Maybe an explicit test for this case could be included alongside `test_format_requirement_ireq_with_hashes`.

There is already a test for this case `test_format_requirement_editable`:
https://github.com/jazzband/pip-tools/blob/57ad5487221d0672f51661997be3b865afe0e19e/tests/test_utils.py#L12-L14

But i think there should be more lines for testing editables and pinned packages with hashes:
https://github.com/jazzband/pip-tools/blob/57ad5487221d0672f51661997be3b865afe0e19e/tests/test_sync.py#L222-L234

Reply:
In that case I'll hold off on merging this.

Reply:
@jdufresne @RDIL i've added couple tests, check it out please guys.